### ISSUE TITLE
Missing dependency to 'path-to-regexp' in core package

### ISSUE THREAD

**Issue by ulrichb** [nestjs/nest#969]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request 
</code></pre>

**Issue:** [fastify-adapter.ts](https://github.com/nestjs/nest/blob/9e6cf1f1aaa4afdbe255d106432e0274b70cbbd1/packages/core/adapters/fastify-adapter.ts#L4) (part of the core package) imports `path-to-regexp`, but the core [package.json](https://github.com/nestjs/nest/blob/9e6cf1f1aaa4afdbe255d106432e0274b70cbbd1/bundle/core/package.json#L7-L14) misses this dependency.

**Workaround:** Manually install `path-to-regexp` in the application.

## Environment
<pre><code>Nest version: 5.1.0

**Comment by kamilmysliwiec** [comment]
It's good for now since Nest explicitly depends on `express` library which in turn depends on `path-to-regexp`. Once we extract `express` to the separated package, we'll get rid of it.

**Comment by ulrichb** [comment]
@kamilmysliwiec The implicit dependency is not enough because npm is allowed to "scope" required modules (e.g. when version conflicts appear).

Here is a repro:
```
npm i -D typescript react-scripts-ts

npm i @nestjs/common @nestjs/core express reflect-metadata rxjs

node Server.js
# => works

npm prune --production
# OR (after deleting node_modules)
npm i --only=prod

node Server.js
# => Error: Cannot find module 'path-to-regexp'
```

... in this situation (when also having `react-scripts-ts` as a dependency, which also requires `path-to-regexp` but in a different version as `express`), the `path-to-regexp` becomes a scoped package within `express`:
![image](https://user-images.githubusercontent.com/388796/44212232-16269500-a16b-11e8-979a-5590d63f507c.png)

 => `fastify-adapter.ts` cannot import `path-to-regexp` as described above. And this is exactly what happens on my prod system.

Full repro sample: [nest-repro-969.zip](https://github.com/nestjs/nest/files/2293859/nest-repro-969.zip)

Further: Shouldn't direct dependencies (and `fastify-adapter.ts` has a direct dependency to `path-to-regexp`) _always_ be expressed as package dependency?  (_Theoretically_ `express` could drop/replace the dependency to `path-to-regexp` ...)

**Comment by kamilmysliwiec** [comment]
Thanks @ulrichb. I'll fix it in `5.2.0` and publish very very soon

**Comment by ulrichb** [comment]
@kamilmysliwiec Okay, great. Many thanks!
